src/library_wasmfs.js

```diff
@@ -272,7 +272,12 @@ FS.createPreloadedFile = FS_createPreloadedFile;
   },
   _wasmfs_copy_preloaded_file_data: function(index, buffer) {
     HEAPU8.set(wasmFSPreloadedFiles[index].fileData, buffer);
-  }
+  },
 });
 
-DEFAULT_LIBRARY_FUNCS_TO_INCLUDE.push('$FS');
+// As mentioned in emcc.py, like the old JS FS, WasmFS by default includes just
+// what JS parts it actually needs, and FORCE_FILESYSTEM is required to force
+// all of it to be included if the user wants to use the JS API directly.
+if (FORCE_FILESYSTEM) {
+  DEFAULT_LIBRARY_FUNCS_TO_INCLUDE.push('$FS');
+}
```

Comment:
This code is shared with the old `library_fs.js` .. can we move it to `src/library_fs_shared.js`?

Reply:
Good idea, moved.